### ISSUE TITLE
cdktf deploy doesnt clearly print errors

### ISSUE THREAD

**Issue by trondhindenes** [hashicorp/terraform-cdk#574]
### Community Note
* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### cdktf & Language Versions
```
╰─ cdktf --version
0.1.0-pre.186
```

### Expected Behavior
it whould be easy to understand what cdktf is doing, and errors should be clearly printed

### Actual Behavior
When dealing with large stacks, cdktf will fill the whole vertical area of the terminal with object names. Any errors will be printed "above", requiring the user to have to scroll up in order to see the errors.

In my opinion, the way cdktf outputs information as it's running is its biggest weak spot. We prefer to hit "no" on the prompt, and actually use regular terraform appl on the generated plan file to get a better sense of what's going on. It's like cdktf tries to be too smart and fails. Simply piping the output from the terraform process would (imho) be better.

### Steps to Reproduce
Deploy a large stack and make it fail somehow :-|

**Comment by trondhindenes** [comment]
Just to add to this: Now that more of our developers have started using cdktf, the feedback is pretty uniform: cdktf is awesome, but the console feedback (e.g. what is it doing, what were the errors etc) is sub-par. For now we've standardized on just going into the `cdktf.out` folder and running terraform commands "directly". Would be good to get a discussion going here.

**Comment by vibe** [comment]
This is definitely a usability concern for new users especially.

To add to the scope of the issue, some terminals, such as the default VSCode terminal behavior will not actually create of cdktf synth/deploy logs but it will try to replace them inline as statuses get updated (I believe it comes from how the cdktf is generated from a `reactjs` powered cli tool - `ink`. I see the benefit in this neat behavior, but the fact errors can be lost in the stream is very bad DX when the summary indicates the deployment failed with errors but said errors are nowhere in sight. 


I workaround it by buffering the output into a file `cdktf deploy --auto-approve > deploy.out` surprisingly this results in the errors being outputted at the very end. 

This seems like an easy enough item to tackle and I would love to volunteer time into a solution but if we can agree on what the appropriate direction is, it will help a lot. 

Things to consider.

0. Do we still want to use inky? It has some cons, (the bane of my existence  is that it does declare it's `react` `devDependency` properly  (and neither do their upstream deps for that matter), which leads cryptic resolution errors, especially in monorepos).
1. Preferred Solutions and prior art?
2. prioritization? 
3. (i thought id come up with more than a couple of things, but it's 4 am my brain is 💩  heh.

**Comment by ansgarm** [comment]
Hi @vibe,
First of all: Sorry for taking so much time for getting back to you!

We are planning to do a refactor of the CLI UI and most importantly the UX. Currently the invocation of the Terraform CLI and all the surrounding steps are coupled quite tightly to React (Hooks) which is something we want to get rid of. In the process of improving that separation we also want to improve the UX of the CLI output. For example showing the changes to the individual attributes of resources in a plan is one of the features in mind.
But the error handling is even more important and should also be part of this, of course.

I think it's unlikely that we're going to drop `ink` altogether in that process. I recently worked some more with it on the new [`cdktf watch` command](https://github.com/hashicorp/terraform-cdk/issues/801) and noticed some of the behaviour you are describing. Currently we print and update all resources to the screen at the same time which cause weird errors when we run out of vertical space, I think there could be a way of using partly dynamic and partly [static](https://github.com/vadimdemedes/ink#static) output to offer a dynamic experience while still preserving errors and history.

For a short term fix for the meantime, I think that the way we print errors could possibly be changed, but I'm just guessing there as I don't know how exactly we print errors (if at all for some 😬). I made some good experience using the `<Static>` component to render things that should be printed once and stay visible, but your hint about using `> deploy.out` as a workaround suggests that maybe that still won't help for some errors.

I hope that I could convey a bit of context on how we think about this. We would be really happy to see you contribute to this particular thing (clearly printing errors), but it could happen that they're short lived and get replaced by a refactor of the UX/UI of the CLI. So I'd rather suggest (if that's something that would interest you), to include your perspective into the design work that's going to happen ahead of the refactor. In particular we are after all the requirements one might expect for a good experience. Such as - of course - showing all errors that happen, or - as mentioned - displaying which attributes will update. If you have some thoughts on this, I'd love to hear them! And/Or as soon as we know more about how we want to tackle this, we'd of course love contributions there as well 🚀

Regarding just a quick fix / improvement: We're happy for any PR that improves this behaviour for the meantime 👍
